src/libstd/keyword_docs.rs

```diff
@@ -1167,12 +1185,31 @@ mod await_keyword {}
 
 #[doc(keyword = "dyn")]
 //
-/// Name the type of a [trait object].
+/// `dyn` is a prefix of a [trait object]'s type.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// The `dyn` keyword is used to highlight that calling methods on a trait is to be achieved by
+/// dynamic dispatch. An instance of a struct passed this way is called a 'trait object' and the
+/// trait must be `object safe` in order to be used in this way.
+///
+/// Unlike generic parameters or `impl Trait`, the compiler does not know the concrete type that
+/// is being passed. As such, a `dyn Trait` reference contains _two_ pointers, the standard one to
```

Comment:
```suggestion
/// is being passed. That is, the type has been [erased].
/// As such, a `dyn Trait` reference contains _two_ pointers.
/// One pointer goes to the data (e.g., an instance of a struct).
/// Another pointer goes to a map of method call names to function pointers
```
(+ link to https://en.wikipedia.org/wiki/Type_erasure)